### PR TITLE
fix: Qr code freeze

### PR DESCRIPTION
**Pull request by Pritam-deb**
issue: https://github.com/atsign-foundation/apps/issues/671

### CODE REVIEW

**Comment by Pritam-deb** [comment]
done and working

File: lib/common_components/qr_scanner.dart [hunk 1]
```diff
@@ -137,8 +138,29 @@ class _QRScannerState extends State<QRScanner> {
                       callback: (container) async {
                         this._controller = container;
                         await _controller!.startCamera((data, offsets) async {
-                          _controller?.stopCamera();
-                          await onScan(data, offsets, context);
+                          //check and make sure that "data" has a valid atsign
+                          bool _atSignValid =
+                              await CommonFunctions().checkAtsign(data);
+                          bool flag;
```
Comment (nitesh2599) [inline lib/common_components/qr_scanner.dart]: @Pritam-deb Is this correct ?

I think this should not pass the null check.

File: lib/common_components/qr_scanner.dart [hunk 2]
```diff
@@ -138,23 +140,34 @@ class _QRScannerState extends State<QRScanner> {
                       callback: (container) async {
```
Comment (yahu1031) [inline lib/common_components/qr_scanner.dart]: Can you please extract this to a menthod. @Pritam-deb

File: lib/common_components/qr_scanner.dart [hunk 3]
```diff
@@ -33,6 +33,31 @@ class _QRScannerState extends State<QRScanner> {
     super.initState();
   }
 
+  scanQR(QrReaderViewController container) async {
```
Comment (yahu1031) [inline lib/common_components/qr_scanner.dart]: Please add the return type too `Future<void>`

File: lib/common_components/qr_scanner.dart [hunk 4]
```diff
@@ -137,32 +162,7 @@ class _QRScannerState extends State<QRScanner> {
...
-                                    ColorConstants.white,
-                                  ),
-                                ),
-                              ));
-                            }
-                            flag = true;
-                          }
-                        });
-                      },
+                      callback: (container) => scanQR(container),
```
Comment (yahu1031) [inline lib/common_components/qr_scanner.dart]: Can you also please rename this callback variable to the appropriate name?

**Review by yahu1031 — CHANGES_REQUESTED** [review]

File: lib/common_components/qr_scanner.dart (hunk 4 above)
Comment (Pritam-deb) [reply lib/common_components/qr_scanner.dart]: I didn't understand. Can you explain?

**Review by yahu1031 — APPROVED** [review]

**Review by sachins-geekyants — APPROVED** [review]

**Pull request merged**